cpp/src/traversal/sssp_kernels.cuh

```diff
@@ -462,7 +469,7 @@ __global__ void relax_edges(
               DistType* update_addr = &next_distances[dst_id];
               DistType old_val = distances[dst_id];
               DistType new_val = distances[src_id] + edge_weights[edge];
-              if (new_val < old_val) {
+              if (new_val <= old_val) {
```

Comment:
This will break sssp for graphs with self loops and 0 weight cycles if you use this information to determine predecessors. Bellman Ford is supposed to work for such graphs and return the shortest simple path.

Reply:
Thank you for pointing it out! I am currently looking into other approaches